category.js

```diff
@@ -0,0 +1,128 @@
+const pathToJsons = "./Jsons/";
+const dropDown = document.getElementById("category");
+const continueBtn = document.getElementById("continue-btn");
+
+function capitalizeCategories(categories) {
+  const array = [];
+  for (let category of categories) {
+    const upperCategory = category[0].toUpperCase() + category.slice(1);
+    array.push(upperCategory);
+  }
+  return array;
+}
+
+function renderData(categories) {
+  const capitalizedCategories = capitalizeCategories(categories);
+  for (let category of capitalizedCategories) {
+    const option = document.createElement("option");
+    option.value = category;
+    option.innerHTML = category;
+    dropDown.appendChild(option);
+  }
+}
+
+async function main() {
+  const categories = await fetchJsonData("categories.json");
+  renderData(categories);
+}
+
+main();
+
+const categories = document.getElementsByClassName("category-button");
+const allButton = document.getElementById("all-button");
+
+const categoryAndIcon = {
+  Clothes: "fa-tshirt",
+  Electronics: "fa-robot",
+  Decoration: "fa-couch",
+  "Sports and Outdoors": "fa-football-ball",
+  Games: "fa-gamepad",
+  Hobbies: "fa-palette",
+  Books: "fa-book-open",
+  Cooking: "fa-cookie-bite",
+  Accessories: "fa-shopping-bag",
+};
+
+function selectOrDeselect(el) {
+  el.classList.toggle("selected");
+}
+
+function selectAll() {
+  for (let category of categories) {
+    category.classList.add("selected");
+  }
+}
+
+function deselectAll() {
+  for (let category of categories) {
+    category.classList.remove("selected");
+  }
+}
+
+function showOrHideIcon(category) {
+  const iconClass = categoryAndIcon[category];
+  const icon = document.getElementsByClassName(iconClass)[0];
+  icon.classList.toggle("invisible");
+}
+
+function showAllIcons() {
+  const icons = document.getElementsByClassName("fas");
+  for (icon of icons) {
+    icon.classList.remove("invisible");
+  }
+}
+
+function hideAllIcons() {
+  const icons = document.getElementsByClassName("fas");
+  for (icon of icons) {
+    icon.classList.add("invisible");
+  }
+}
+
+for (let category of categories) {
+  category.addEventListener("click", () => {
+    selectOrDeselect(category);
+    showOrHideIcon(category.innerHTML);
+    allButton.classList.remove("black-selected");
+  });
+}
+
+allButton.addEventListener("click", () => {
+  allButton.classList.toggle("black-selected");
+  if (allButton.classList.contains("black-selected")) {
+    selectAll();
+    showAllIcons();
+  } else {
+    deselectAll();
+    hideAllIcons();
+  }
+});
+
+const dropDownDiv = document.getElementById("select-div");
+let cookieValue = [];
+
+function checkWhatCookieValue() {
```

Comment:
**Question**: What is this function trying to do?🤔 
I am a bit confused since it says that it is checking some cookie value but we don't pass anything as an argument 😅

Reply:
Since on mobile there is a drop down and on the computer there are buttons, this checks if the cookie value should be of the drop down or of the buttons. It also makes it so that if they don't pick anything it defaults to "all". (sorry I never know what to name things 😅)